### PR TITLE
Homework4

### PR DESCRIPTION
**Pull request by darhan1168**
Did homework:
- Create a project (class library) that will automatically map property values between models and also supports the ability to create specific mappings.
- Create a project (class library) that will allow you to create your own validators for models that will be called using the object.Validate() extension method. (the method will also need to be created)

### CODE REVIEW

File: Homework4/Homework4/Validators.cs [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+        foreach (var validator in _validators)
+        {
+            if (!validator.Validate(model))
+            {
+                return false;
+            }
+        }
+
+        return true;
+    }
```
Comment (vemontd1) [inline Homework4/Homework4/Validators.cs]: You can enhance this by returning custom ValidationResult, so you can collect list of exceptions or error messages there

File: Homework4/Homework4/LibraryForMapping.cs [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+        foreach (var sourceProperty in sourceProperties)
+        {
+            var destinationProperty = destinationProperties.FirstOrDefault(x => x.Name == sourceProperty.Name);
+            if (destinationProperty != null)
+            {
+                destinationProperty.SetValue(destination, sourceProperty.GetValue(source));
+            }
+        }
+    }
```
Comment (vemontd1) [inline Homework4/Homework4/LibraryForMapping.cs:18]: Can you also add addition methods to map only specific properties?

**Review by vemontd1 — CHANGES_REQUESTED** [review]
Good job overall

File: Homework4/Homework4/Validators.cs (hunk 1 above)
Comment (darhan1168) [reply Homework4/Homework4/Validators.cs]: Fixed

File: Homework4/Homework4/LibraryForMapping.cs (hunk 2 above)
Comment (darhan1168) [reply Homework4/Homework4/LibraryForMapping.cs:18]: Fixed

File: Homework4/Homework4/Validators.cs [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+    public ValidationResult Validate(T model)
+    {
+        var result = new ValidationResult();
+
+        foreach (var validator in _validators)
+        {
+            try
+            {
+                validator.Validate(model);
```
Comment (JustDimus) [inline Homework4/Homework4/Validators.cs:22]: IValidator<T>.Validate method returns a bool value. Where do you use it?

**Review by JustDimus — APPROVED** [review]

File: Homework4/Homework4/Validators.cs (hunk 3 above)
Comment (darhan1168) [reply Homework4/Homework4/Validators.cs:22]: Fixed

**Pull request merged**